libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,235 @@
+
+            // Configuration
+            string applicationRoot = configuration.GetSection(ConfigurationConstants.ApplicationRootKey).Value;
+            string defaultLocale = configuration.GetSection(ConfigurationConstants.DefaultLocale).Value;
+            string rootDialog = configuration.GetSection(ConfigurationConstants.RootDialogKey).Value;
+            
+            // Runtime settings. If no config is provided, we create basic runtime config with defaults.
+            var runtimeSettings = configuration.GetSection(ConfigurationConstants.RuntimeSettingsKey).Get<RuntimeSettings>() ?? new RuntimeSettings();
+
+            // Bot
+            services.AddSingleton<IBot, CoreBot>();
+            services.AddOptions()
+                .Configure<CoreBotOptions>(o =>
+                {
+                    o.DefaultLocale = defaultLocale;
+                    o.RootDialog = rootDialog;
+                });
+
+            services.AddSingleton(configuration);
+
+            // ResourceExplorer. TryAddSingleton will only add if there is no other registration for resource explorer.
+            // Tests use this to inject custom resource explorers but could also be used for advanced runtime customization scenarios.
+            services.TryAddSingleton<ResourceExplorer>(serviceProvider =>
+                new ResourceExplorer()
+                    .AddFolder(applicationRoot)
+                    .RegisterType<OnQnAMatch>(OnQnAMatch.Kind));
+            
+            // Runtime set up
+            services.AddBotRuntimeSkills(runtimeSettings.Skills);
+            services.AddBotRuntimeStorage(configuration, runtimeSettings);
+            services.AddBotRuntimeTelemetry(runtimeSettings.Telemetry);
+            services.AddBotRuntimeTranscriptLogging(configuration, runtimeSettings.Features);
+            services.AddBotRuntimeFeatures(runtimeSettings.Features);
+            services.AddBotRuntimePlugins(configuration, runtimeSettings);
+            services.AddBotRuntimeAdapters(runtimeSettings);
+        }
+
+        internal static void AddBotRuntimeSkills(this IServiceCollection services, SkillSettings skillSettings)
+        {
+            services.AddSingleton(sp => new AuthenticationConfiguration { ClaimsValidator = new AllowedCallersClaimsValidator(skillSettings?.AllowedCallers) });
+            services.AddSingleton<SkillConversationIdFactoryBase, SkillConversationIdFactory>();
+            services.AddHttpClient<BotFrameworkClient, SkillHttpClient>();
+            services.AddSingleton<ChannelServiceHandler, SkillHandler>();
+        }
+
+        internal static void AddBotRuntimeStorage(this IServiceCollection services, IConfiguration configuration, RuntimeSettings runtimeSettings)
+        {
+            services.AddSingleton<UserState>();
+            services.AddSingleton<ConversationState>();
+
+            // Cosmosdb
+            if (runtimeSettings?.Storage == typeof(CosmosDbPartitionedStorage).Name)
```

Comment:
Use nameof(type) instead of typeof(type).Name (there are several instances in this file.
Nameof is slightly more performant (see for https://stackoverflow.com/questions/33649501/difference-between-nameof-and-typeof more details)

Reply:
BTW, this is the same issue @mrivera-ms raised and it should work use:
```suggestion
            if (runtimeSettings?.Storage == nameof(CosmosDbPartitionedStorage))
```

Reply:
Yes I had completely misunderstood. Updating.